fix: mixed-type scenarios in lucene range expressions (#1019)
fkie-cad/Logprep: CHANGELOG.md

### Breaking
* restrict UNIX timestamp normalization to seconds, milliseconds, microseconds, and nanoseconds.
* handle non-string `concatenator` source fields with `ProcessingWarning` instead of `ProcessingCriticalError`
* filter: allow mixed numeric range boundaries and type coercion for range matching

> * filter: unquoted asterisks (`*`) in range expressions are now considered open boundaries

Reviewer: `* filter: unquoted asterisks (`*`) in range expressions are now considered open boundaries`

I'don't think this belongs under **Breaking** and can be removed. Previously, open `*` boundaries were rejected, while this change adds support for them. That seems like a new feature rather than a breaking change, and it is already covered under **Features** below.

### Breaking
* restrict UNIX timestamp normalization to seconds, milliseconds, microseconds, and nanoseconds.
* handle non-string `concatenator` source fields with `ProcessingWarning` instead of `ProcessingCriticalError`

> * filter: rework range expressions to allow mixed types and type coercion

Reviewer: ```suggestion
* filter: allow mixed numeric range boundaries and type coercion for range matching
```

> * filter: drop support for `inf`/`nan` in range expressions

Reviewer: ```suggestion
* filter: treat `inf`/`nan` as string values instead of numeric range boundaries
```

and I'm not sure whether `inf`/`nan` should be listed under **Breaking** at all.

Previously, these expressions were explicitly rejected as invalid with `Range boundaries must be finite numbers`. With this change, they are accepted and fall back to lexicographic string comparison instead.

So the current changelog entry `drop support for inf/nan` seems misleading. Support isn't being dropped; previously invalid expressions are becoming valid with different semantics.

Author: Fair enough. So it's rather a bugfix as these are implementation details which leaked the abstraction before

> * filter: add support for `*` (open boundary) in range expressions

Reviewer: Breaking or Feature?
